backend/core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/ShowFixSuggestionRequestHandler.java

```diff
@@ -0,0 +1,376 @@
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.initialize.InitializeParams;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.connection.AssistCreatingConnectionParams;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.connection.SonarCloudConnectionParams;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.connection.SonarQubeConnectionParams;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.fix.ChangesDto;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.fix.FileEditDto;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.fix.FixSuggestionDto;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.fix.LineRangeDto;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.fix.ShowFixSuggestionParams;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.message.MessageType;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.message.ShowMessageParams;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.telemetry.FixSuggestionReceivedParams;
+import org.sonarsource.sonarlint.core.telemetry.TelemetryService;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+
+@Named
+@Singleton
+public class ShowFixSuggestionRequestHandler implements HttpRequestHandler {
+
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+
+  private final SonarLintRpcClient client;
+  private final TelemetryService telemetryService;
+  private final SonarProjectBranchTrackingService branchTrackingService;
+  private final boolean canOpenFixSuggestion;
+  private final RequestHandlerBindingAssistant requestHandlerBindingAssistant;
+  private final PathTranslationService pathTranslationService;
+  private final String sonarCloudUrl;
+
+  public ShowFixSuggestionRequestHandler(SonarLintRpcClient client, TelemetryService telemetryService, SonarProjectBranchTrackingService branchTrackingService,
+    InitializeParams params, RequestHandlerBindingAssistant requestHandlerBindingAssistant,
+    PathTranslationService pathTranslationService, SonarCloudActiveEnvironment sonarCloudActiveEnvironment) {
+    this.client = client;
+    this.telemetryService = telemetryService;
+    this.branchTrackingService = branchTrackingService;
+    this.canOpenFixSuggestion = params.getFeatureFlags().canOpenFixSuggestion();
+    this.requestHandlerBindingAssistant = requestHandlerBindingAssistant;
+    this.pathTranslationService = pathTranslationService;
+    this.sonarCloudUrl = sonarCloudActiveEnvironment.getUri().toString();
+  }
+
+  @Override
+  public void handle(ClassicHttpRequest request, ClassicHttpResponse response, HttpContext context) throws HttpException, IOException {
+    var showFixSuggestionQuery = extractQuery(request);
+    if (!canOpenFixSuggestion || !Method.POST.isSame(request.getMethod()) || !showFixSuggestionQuery.isValid()) {
```

Comment:
Regarding the second condition `!Method.POST.isSame(request.getMethod())`, we control the server side, so I'm not sure we need to be that defensive here.

Reply:
I'd rather keep it, I don't see a reason not to